### PR TITLE
Balance module validation forms

### PR DESCRIPTION
**Pull request by fabriziovigevani**
Includes:

- All error validations for the ANJ balance interactions
- Balances formatted to 2 decimals
- Input to set the max amount.
- Fixed error that set the max decimals for the amount to 3

### CODE REVIEW

File: src/components/Dashboard/panels/ANJForm.js [hunk 1]
```diff
@@ -1,38 +1,49 @@
...
-        digits: Math.min(anjToken.decimals, MAX_INPUT_DECIMAL_BASE),
-      }),
-    [anjToken.decimals]
-  )
-
   // Change amount handler
   const handleAmountChange = useCallback(event => {
     const newAmount = event.target.value
     setAmount(amount => ({ ...amount, value: newAmount }))
   }, [])
```
Comment (bpierre) [inline src/components/Dashboard/panels/ANJForm.js]: I think we should store two values here: `event.target.value` to update the field in the next render, and a filtered version of it as a BN.js number.

If the value can’t be converted into a BN.js number, we could have a value representing its impossible state. For example, if the user types “abc”, we could use `bigNum(-1)` or `null`.

Another option would be to do this on submit, but we might want to know the parsed value before that (e.g. to inform that the value is higher than the max).

File: src/components/Dashboard/panels/ANJForm.js [hunk 2]
```diff
@@ -56,21 +69,53 @@ const ANJForm = React.memo(function ANJForm({
...
           name="amount"
           wide
           onChange={handleAmountChange}
           value={amount.value}
-          step={minStep}
-          min="1"
           ref={inputRef}
           required
+          adornment={
+            <span
```
[the remark this answers is not in the document]
Comment (fabriziovigevani) [reply src/components/Dashboard/panels/ANJForm.js]: Ah yes good call

File: src/lib/math-utils.js [hunk 3]
```diff
@@ -38,7 +38,7 @@ export function parseUnits(value, digits) {
  */
 export function formatUnits(
   value,
-  { digits = 18, commas = true, replaceZeroBy = '0' } = {}
+  { digits = 18, commas = true, replaceZeroBy = '0', precision = 2 } = {}
```
[the remark this answers is not in the document]
Comment (fabriziovigevani) [reply src/lib/math-utils.js]: Totally missed it 😅

File: src/components/Dashboard/panels/ANJForm.js (hunk 1 above)
Comment (fabriziovigevani) [reply src/components/Dashboard/panels/ANJForm.js]: Ah yes, good idea. I also ended up using the common form errors on the `ANJForm` and removing them from the parent panel.

File: src/components/Dashboard/panels/ANJForm.js [hunk 4]
```diff
@@ -41,14 +59,28 @@ const ANJForm = React.memo(function ANJForm({
         commas: false,
         precision: anjToken.decimals,
       }),
+      valueBN: maxAmount,
     }))
   }, [anjToken.decimals, maxAmount])
 
+  // Form valdiation
```
Comment (bpierre) [inline src/components/Dashboard/panels/ANJForm.js]:
```suggestion
  // Form validation
```

**Pull request merged**